### ISSUE TITLE
SLA Broadsoft problems in opensips 2.4?

### ISSUE THREAD

**Issue by harishyadav1465** [OpenSIPS/opensips#1578]
Hello,
   I have been testing the **Broadsoft SLA** on the opensips,however it is not working as expected.I am using the **presence_callinfo** module alone for this feature, I made the changes according to the **presence_callinfo** module documentation also  have seen **b2b_sca** ,i think this module is  not mandatory for Brodsoft SCA.

  The problem with **presence_callinfo** module is subscribers of **call-info** Event are not notifying whenever the dialog is in **hold** state..I have gone through the code ,somehow **sca_dialog_callback** is not calling from the **dailog** module whenever the call is in hold state.

 is there anything i missed ?

Thank you

**Comment by razvancrainea** [comment]
You are right, the current code does not publish *hold* states for the dialog. I will mark this as a feature request.

**Comment by harishyadav1465** [comment]
Hello,
           Thanks for the reply.
           I have changed the code for **hold state**,it is working as expected but when the **watcher** tries to pick the call it not **bridging** to the old call..take this also a **feature request**.
           Besides,I have gone through the **b2b_sca** module and  set the configuration file according to it ,however it is working as expected for the **incoming call** to the **SHARED** line but not the **outgoing call** from the **SHARED** line. 

 Is there anything i am missing here?           

Thank you

**Comment by bogdan-iancu** [comment]
Hey @harishyadav1465 , could you share you changes for the 'hold' support ?

In terms of  call pick up, this is something outside the scope of the presence_callinfo module - the module just publishes the line state, it is not performing any call related actions. As you said, I guess you made the b2b scenario in order to do this.

Maybe posting pcap (from opensips server) to see the call and presence signaling will help
